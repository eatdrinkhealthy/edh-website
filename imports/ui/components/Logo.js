import React from "react";
import { logoStyle } from "../styles";

const Logo = () => (
  <div className={logoStyle.className} />
);

export default Logo;
