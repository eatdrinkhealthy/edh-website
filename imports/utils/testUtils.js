/* eslint-env jest */
/* eslint-disable func-names, prefer-arrow-callback, no-unused-expressions */
/* eslint-disable import/no-extraneous-dependencies */

import cheerio from "cheerio";
import _ from "lodash";
import {
  mount,
} from "enzyme";

export function getComponentClasses(currComponent) {
  // NOTE: using mount here to ensure all child elements are fully rendered
  //       but is seems shallow might work too
  const mountedHTML = mount(currComponent).html();
  console.log("html:", mountedHTML);

  const $ = cheerio.load(mountedHTML);
  let classList = [];
  $("*").each(function () {
    // add new unique classes of any element
    classList = _.union(classList, $(this).attr("class").split(" "));
  });
  return classList;
}
