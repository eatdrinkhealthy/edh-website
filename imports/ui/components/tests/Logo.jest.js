/* eslint-env jest */
/* eslint-disable func-names, prefer-arrow-callback, no-unused-expressions */

import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Logo from "../Logo";
import { formattedJSON } from "../../../utils/tests/testUtils";

describe("<Logo />", function () {
  it("render matches snapshot", function () {
    const tree = renderer.create(<Logo />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("css matches snapshot", function () {
    expect(shallow(<Logo />).hasClass(Logo.style.className)).toBe(true);
    expect(formattedJSON(Logo.style)).toMatchSnapshot();
  });
});
