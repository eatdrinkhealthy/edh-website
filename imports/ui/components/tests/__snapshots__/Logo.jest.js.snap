exports[`<Logo /> matches render snapshot 1`] = `
<div
  className="css-o94tcm" />
`;

exports[`<Logo /> matches style / css snapshot 1`] = `
"[
  {
    \"rules\": {
      \"backgroundImage\": \"url(\'/images/ios-icon.png\')\",
      \"backgroundSize\": \"100%\",
      \"width\": \"90px\",
      \"height\": \"90px\",
      \"margin\": \"30px auto\"
    },
    \"className\": \"css-o94tcm\"
  }
]"
`;
